Library/include/CSP/Multiplayer/Components/TextSpaceComponent.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// @file TextSpaceComponent.h
+/// @brief Definitions and support for text components.
+
+#pragma once
+
+#include "CSP/CSPCommon.h"
+#include "CSP/Common/String.h"
+#include "CSP/Multiplayer/ComponentBase.h"
+#include "CSP/Multiplayer/Components/BillBoardModeEnum.h"
+#include "CSP/Multiplayer/Components/Interfaces/ITransformComponent.h"
+#include "CSP/Multiplayer/Components/Interfaces/IVisibleComponent.h"
+
+
+namespace csp::multiplayer
+{
+
+/// @brief Enumerates the list of properties that can be replicated for an text space component.
+enum class TextPropertyKeys
+{
+	Position,
+	Rotation,
+	Scale,
+	Text,
+	TextColor,
+	BackgroundColor,
+	IsBackgroundVisible,
+	Width,
+	Height,
+	BillboardMode,
+	IsVisible,
+	IsARVisible,
+	Num
+};
+
+
+/// @ingroup TextSpaceComponent
+/// @brief Data representation of an TextSpaceComponent.
+class CSP_API TextSpaceComponent : public ComponentBase, public ITransformComponent, public IVisibleComponent
+{
+public:
+	/// @brief Constructs the text space component, and associates it with the specified Parent space entity.
+	/// @param Parent The Space entity that owns this component.
+	TextSpaceComponent(SpaceEntity* Parent);
+
+	/// @brief Gets the text  this text component refers to.
+	/// @return The text this text component refers to.
+	const csp::common::String& GetText() const;
+
+	/// @brief Sets the text this text component refers to.
+	/// @param Value The text this text component refers to.
+	void SetText(const csp::common::String& Value);
+
+	/// @brief Gets the text color.
+	/// @return The text color.
+	const csp::common::Vector3& GetTextColor() const;
```

Comment:
Can we please have more details on the format of the colour similarly to how GaussianSplatComponent details the tint colour ? It would definitely help with integration.

Example from GaussianSplatComponent
`/// @brief Sets the tint that should be globally applied to the Gaussian Splat.
	/// @param Value The tint value, expected to be in RGB color space, with each value normalised between 0...1.
	/// Defaults to 1,1,1.`

Reply:
Addressed in latest 👍